bindings/python/scripts/parse.py

```diff
@@ -1,145 +1,143 @@
-import os
-import argparse
 import sys
-import json
-
 import clang.cindex as clang
 
+import utils
 
-class bcolors:
-    HEADER = "\033[95m"
-    OKBLUE = "\033[94m"
-    OKGREEN = "\033[92m"
-    WARNING = "\033[93m"
-    FAIL = "\033[91m"
-    ENDC = "\033[0m"
-    BOLD = "\033[1m"
-    UNDERLINE = "\033[4m"
 
+def is_node_in_this_file(cursor, filename):
+    """
+    Checks if the node in the AST belongs to the file
+    
+    Arguments:
+        - cursor : The cursor pointing to the node
+        - filename : The file's name to check the node against
+        
+    Returns:
+        - True/False (bool)
+    """
 
-def print_node(cursor, lines, depth):
-    file = cursor.location.file
-    line, column = cursor.location.line, cursor.location.column
-    result_type = (
-        cursor.result_type.spelling if cursor.result_type.spelling is not "" else "None"
-    )
-    print(
-        "-" * depth,
-        file,
-        f"L{line} C{column}",
-        bcolors.BOLD + cursor.kind.name + bcolors.ENDC,
-        bcolors.OKBLUE + cursor.spelling + bcolors.ENDC,
-        bcolors.OKGREEN + cursor.access_specifier.name + bcolors.ENDC,
-        bcolors.OKGREEN + "->" + bcolors.ENDC,
-        bcolors.OKGREEN + result_type + bcolors.ENDC,
-    )
+    if cursor.location.file and cursor.location.file.name == filename:
+        return True
+    else:
+        return False
 
 
-def node_in_this_file(node, file_name):
-    return node.location.file and node.location.file.name == file_name
+def print_ast(cursor, this_filename, depth):
+    """
+    Prints the AST by recursively traversing the AST
 
+    Arguments:
+        - cursor: The cursor pointing to a node
+        - this_filename: The file's name to check if the node belongs to it
+        - depth: The depth of the node (root=0)
+    
+    Returns:
+        - None
+    """
 
-def walk_and_print(cursor, filter, lines, this_filename, depth):
     if cursor.spelling:
-        print_node(cursor, lines, depth)
+        print(
+            "-" * depth,
+            cursor.location.file,
+            f"L{cursor.location.line} C{cursor.location.column}",
+            cursor.kind.name,
+            cursor.spelling,
+        )
 
     for child in cursor.get_children():
-        if node_in_this_file(child, this_filename):
-            walk_and_print(child, filter, lines, this_filename, depth + 1)
+        if is_node_in_this_file(cursor=child, filename=this_filename):
+            print_ast(cursor=child, this_filename=this_filename, depth=depth + 1)
+
+
+def generate_parsed_info(cursor, this_filename, depth):
```

Comment:
`cursor` and `this_filename` seem to go together. Maybe make a struct, tuple or something to make them go together.

This is also because the cursor needs to point in to the file requested (which you aren't checking at the beginning of the function)

Reply:
> Maybe make a struct, tuple or something to make them go together.

Done.

 > which you aren't checking at the beginning of the function

I tried, but the root (`TRANSLATION_UNIT`) has `cursor.location.file` as NoneType, so it leads to an empty json.